syncer/syncer.go

```diff
@@ -292,7 +292,6 @@ func (s *Syncer) synchronize(ctx context.Context) bool {
 			logger.Info("setting layer %v to zero-block", layerID)
 			if err := s.mesh.SetZeroBlockLayer(layerID); err != nil {
```

Comment:
I think we shouldn't fail sync if there are already blocks in the layer... I wonder if this can happen now after your previous fix

Reply:
i think it's still possible in the case where
- we received a late block before our peers
- the peer we connect to is not fully synced, and other peers happen to return error
the chance is there but small. either way, like you said, i think we still should not fail sync for this.